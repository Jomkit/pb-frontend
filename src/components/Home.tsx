import Frame from './Frame'
import TaskSelectable from '../features/task/TaskSelectable'
import Timer from '../features/timer/Timer';
import Card from './Card';
import { Link } from 'react-router-dom';
import userContext from './contexts/userContext';
import { useContext } from 'react';

const Home = () => {
  const user = useContext(userContext);

  return (
    <Frame bgColor="bg-cyan-500" title="Productivity Buddy">
      { user.id ? <TaskSelectable /> : 
        <>
        <Card title="Welcome!">
          <div>
            <p>Productivity Buddy is a timer app for students and people who want to get work done</p>
            <p>To access the main features, please <Link to="/login" className='text-blue-500 underline'>login</Link> or <Link to="/signup" className='text-blue-500 underline'>signup</Link></p>
            <p>Or, start focusing now with a timer!</p>
          </div>

        </Card>
          <Timer />
          <p>*Note, data will not be recorded without logging in</p>
        </>
      }
    </Frame>
  )
}

export default Home