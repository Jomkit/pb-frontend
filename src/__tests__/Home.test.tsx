import { render } from "@testing-library/react";
import Home from "../components/Home";
import UserContext from "../components/contexts/userContext";
import { MemoryRouter } from "react-router-dom";

describe("Home component", function() {
    vi.mock("../api");
    
    const testUser = {
        username: "testUser",
        id: 1,
        firstName: "test",
        lastName: "user",
        email: "test@user"
    }

    it('should render', () => {
        render(
            <MemoryRouter>
                <Home />
            </MemoryRouter>
        );
    })

    it("should match snapshot of no logged-in user", () => {
        const { asFragment } = render(
            <MemoryRouter>
                <Home />
            </MemoryRouter>
        )
        expect(asFragment()).toMatchSnapshot();
    })
    
    it("should match snapshot of logged-in user before loading", () => {
        const { asFragment } = render(
            <MemoryRouter>
                <UserContext.Provider value={testUser}>
                    <Home />
                </UserContext.Provider>
            </MemoryRouter>
        )

        expect(asFragment()).toMatchSnapshot();
    })

    it("should match snapshot of logged-in user after loading", async () => {
        const { asFragment, findByText } = render(
            <MemoryRouter>
                <UserContext.Provider value={testUser}>
                    <Home />
                </UserContext.Provider>
            </MemoryRouter>
        )
        
        const addTaskBtn = await findByText("New Task");
        const Timer = await findByText("Timer");
        expect(addTaskBtn).toBeInTheDocument();
        expect(Timer).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    })
})