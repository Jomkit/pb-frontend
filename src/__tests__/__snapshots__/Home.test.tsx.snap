// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component > should match snapshot of logged-in user after loading 1`] = `
<DocumentFragment>
  <section
    class="text-center items-start min-h-screen w-screen p-4 bg-cyan-500"
  >
    <h1>
      Productivity Buddy
    </h1>
    <div
      class="p-10 justify-center"
    >
      <h2
        class="pb-6"
      >
        Let's get to work!
      </h2>
      <div
        class="simple-border rounded w-fit bg-slate-500 h-6"
      >
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          class="group relative flex items-stretch justify-center p-0.5 text-center font-medium transition-[color,background-color,border-color,text-decoration-color,fill,stroke,box-shadow] focus:z-10 focus:outline-none border border-transparent bg-cyan-700 text-white focus:ring-4 focus:ring-cyan-300 enabled:hover:bg-cyan-800 dark:bg-cyan-600 dark:focus:ring-cyan-800 dark:enabled:hover:bg-cyan-700 rounded-lg w-fit"
          data-testid="flowbite-dropdown-target"
          id=":r1:"
          type="button"
        >
          <span
            class="flex items-stretch transition-all duration-200 rounded-md px-4 py-2 text-sm"
          >
            tasks
            <svg
              aria-hidden="true"
              class="ml-2 h-4 w-4"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 9l-7 7-7-7"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        class="w-full lg:w-3/4 m-auto"
      >
        <div
          class="text-start"
        >
          <button
            class="btn bg-green-400 border-solid border-2 border-black"
          >
            New Task
          </button>
        </div>
        <div
          class="w-full mx-auto mb-2 p-6 bg-white border-solid border-2 border-black rounded-lg divide-y"
        >
          <div
            class="text-center"
          >
            <h2>
              testTask 1
            </h2>
            <h4>
              Project: Unassigned
            </h4>
          </div>
          <a
            href="/tasks/1"
          >
            <p
              class="btn bg-blue-500 text-white"
            >
              Click for more info
            </p>
          </a>
          <div
            class="m-2"
          >
            <div
              class="w-full mx-auto mb-2 p-6 bg-white border-solid border-2 border-black rounded-lg divide-y"
            >
              <div
                class="text-center"
              >
                <h2 />
                <h4 />
              </div>
              <h2
                class="text-start"
              >
                Timer
              </h2>
              <div
                class="flex justify-between"
              >
                <div
                  class="text-center"
                >
                  <h3
                    class="text-2xl bg-slate-200 w-fit m-auto px-4 py-1 rounded border-2 border-slate-500"
                  >
                    25:00
                  </h3>
                </div>
                <div
                  id="set-timer"
                >
                  <button
                    class="btn bg-green-400"
                    id="start-btn"
                  >
                    <i
                      class="fa-solid fa-play"
                    />
                  </button>
                  <button
                    class="btn bg-red-500"
                    id="restart-btn"
                  >
                    <i
                      class="fa-solid fa-rotate-right"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Home component > should match snapshot of logged-in user before loading 1`] = `
<DocumentFragment>
  <section
    class="text-center items-start min-h-screen w-screen p-4 bg-cyan-500"
  >
    <h1>
      Productivity Buddy
    </h1>
    <div
      class="p-10 justify-center"
    >
      <h2
        class="pb-6"
      >
        Let's get to work!
      </h2>
      <p>
        Loading...
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Home component > should match snapshot of no logged-in user 1`] = `
<DocumentFragment>
  <section
    class="text-center items-start min-h-screen w-screen p-4 bg-cyan-500"
  >
    <h1>
      Productivity Buddy
    </h1>
    <div
      class="p-10 justify-center"
    >
      <div
        class="w-full mx-auto mb-2 p-6 bg-white border-solid border-2 border-black rounded-lg divide-y"
      >
        <div
          class="text-center"
        >
          <h2>
            Welcome!
          </h2>
          <h4 />
        </div>
        <div>
          <p>
            Productivity Buddy is a timer app for students and people who want to get work done
          </p>
          <p>
            To access the main features, please 
            <a
              class="text-blue-500 underline"
              href="/login"
            >
              login
            </a>
             or 
            <a
              class="text-blue-500 underline"
              href="/signup"
            >
              signup
            </a>
          </p>
          <p>
            Or, start focusing now with a timer!
          </p>
        </div>
      </div>
      <div
        class="w-full mx-auto mb-2 p-6 bg-white border-solid border-2 border-black rounded-lg divide-y"
      >
        <div
          class="text-center"
        >
          <h2 />
          <h4 />
        </div>
        <h2
          class="text-start"
        >
          Timer
        </h2>
        <div
          class="flex justify-between"
        >
          <div
            class="text-center"
          >
            <h3
              class="text-2xl bg-slate-200 w-fit m-auto px-4 py-1 rounded border-2 border-slate-500"
            >
              25:00
            </h3>
          </div>
          <div
            id="set-timer"
          >
            <button
              class="btn bg-green-400"
              id="start-btn"
            >
              <i
                class="fa-solid fa-play"
              />
            </button>
            <button
              class="btn bg-red-500"
              id="restart-btn"
            >
              <i
                class="fa-solid fa-rotate-right"
              />
            </button>
          </div>
        </div>
      </div>
      <p>
        *Note, data will not be recorded without logging in
      </p>
    </div>
  </section>
</DocumentFragment>
`;
